web/src/zform.js

```diff
@@ -1,43 +1,68 @@
 import $ from "jquery";
+import {z} from "zod";
 
 import render_widgets_zform_choices from "../templates/widgets/zform_choices.hbs";
 
 import * as blueslip from "./blueslip";
-import * as schema from "./schema";
 import * as transmit from "./transmit";
 
 export function validate_extra_data(data) {
-    function check(data) {
-        function check_choice_data(data) {
-            function check_choice_item(field_name, val) {
-                return schema.check_record(field_name, val, {
-                    short_name: schema.check_string,
-                    long_name: schema.check_string,
-                    reply: schema.check_string,
-                });
-            }
-
-            function check_choices(field_name, val) {
-                return schema.check_array(field_name, val, check_choice_item);
-            }
-
-            return schema.check_record("zform data", data, {
-                heading: schema.check_string,
-                choices: check_choices,
-            });
-        }
-
-        if (data.type === "choices") {
-            return check_choice_data(data);
-        }
-
-        return "unknown zform type: " + data.type;
+    if (data.type !== "choices") {
+        blueslip.warn(`unknown zform type: ${data.type}`);
+        return false;
     }
 
-    const msg = check(data);
-
-    if (msg) {
-        blueslip.warn(msg);
+    const zFormSchema = z.object(
+        {
+            choices: z.array(
+                z.object(
+                    {
+                        long_name: z.string({
+                            invalid_type_error: "long_name is not a string",
```

Comment:
As [previously discussed](https://chat.zulip.org/#narrow/stream/6-frontend/topic/typescript.20migration/near/1545529), we should not be overriding the errors from Zod. These errors are for developers and should be left standard, direct, and unfiltered. Just use the Zod default error messages.

Reply:
Kayy